packages/client/src/runtime/utils/serializeRawParameters.ts

```diff
@@ -1,72 +1,113 @@
+    return {
+      prisma__type: 'bigint',
+      prisma__value: parameter.toString(),
+    }
+  }
+
+  if (isDate(parameter)) {
     return {
       prisma__type: 'date',
-      prisma__value: data.toJSON(),
+      prisma__value: parameter.toJSON(),
     }
   }
 
-  if (type === '[object Object]') {
-    const tmp = {}
-    for (const key in data) {
-      if (key !== '__proto__') {
-        tmp[key] = replaceDates(data[key])
-      }
+  if (Decimal.isDecimal(parameter)) {
+    return {
+      prisma__type: 'decimal',
+      prisma__value: parameter.toJSON(),
     }
-    return tmp
   }
 
-  if (type === '[object Array]') {
-    let k = data.length
-    let tmp
-    for (tmp = new Array(k); k--; ) {
-      tmp[k] = replaceDates(data[k])
+  if (Buffer.isBuffer(parameter)) {
+    return {
+      prisma__type: 'bytes',
+      prisma__value: parameter.toString('base64'),
     }
-    return tmp
   }
 
-  return data
+  if (isArrayBufferLike(parameter) || ArrayBuffer.isView(parameter)) {
+    return {
+      prisma__type: 'bytes',
+      prisma__value: Buffer.from(parameter).toString('base64'),
+    }
+  }
+
+  if (typeof parameter === 'object' && objectSerialization === 'slow') {
+    return preprocessObject(parameter)
+  }
+
+  return parameter
+}
+
+function isDate(value: any): value is Date {
+  if (value instanceof Date) {
+    return true
+  }
+
+  // Support dates created in another V8 context
+  // Note: dates don't have Symbol.toStringTag defined
+  return Object.prototype.toString.call(value) === '[object Date]' && typeof value.toJSON === 'function'
 }
 
-/**
- * Serializes BigInt as a string https://github.com/prisma/prisma/issues/5823
- * @param data parameters
- */
-export function serializeBigInt(data: any): any {
-  const type = Object.prototype.toString.call(data)
+function isArrayBufferLike(value: any): value is ArrayBufferLike {
+  if (value instanceof ArrayBuffer || value instanceof SharedArrayBuffer) {
+    return true
+  }
 
-  if (type === '[object BigInt]') {
-    return data.toString()
+  if (typeof value === 'object' && value !== null) {
+    return value[Symbol.toStringTag] === 'ArrayBuffer' || value[Symbol.toStringTag] === 'SharedArrayBuffer'
   }
 
-  if (type === '[object Object]') {
-    const tmp = {}
-    for (const key in data) {
-      if (key !== '__proto__') {
-        tmp[key] = serializeBigInt(data[key])
-      }
-    }
-    return tmp
+  return false
+}
+
+function preprocessObject(obj: any): unknown {
+  if (typeof obj !== 'object' || obj === null) {
+    return obj
   }
 
-  if (type === '[object Array]') {
-    let k = data.length
-    let tmp
-    for (tmp = new Array(k); k--; ) {
-      tmp[k] = serializeBigInt(data[k])
-    }
-    return tmp
+  if (typeof obj.toJSON === 'function') {
+    return obj.toJSON()
+  }
+
+  // TODO: map from blaze would've been convenient here to map arrays and objects uniformly,
+  // but importing it in this file causes compilation errors in tsd tests.
```

Comment:
Yes, we need to make it a dependency/package at some point.

Reply:
@aqrln Seems like all works with `tsd` after the latest upgrade. At least I can't reproduce the error locally. All worked after bringing back the import how it was done by you in one of commits.

I wasn’t brave enough to open a PR, because my relationship with Docker is bad. I went through setup steps in Contributing Guide few times, but few images do not start so I can’t see if all tests are passing.

In case if the error persist, I would be curious to see how it looks like. Interesting case (;

Reply:
@mrazauskas thanks, that's good to know! You can still open a PR if you'd like to — the tests will run on CI anyway, so it's not critical to run them locally.

As for images that did not start, I assume you might have had a problem with SQL Server if you're on ARM64?